nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -0,0 +1,176 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let
+  cfg = config.services.lemmy;
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options.services.lemmy = {
+    enable = mkEnableOption "lemmy server";
+
+    useLocalPostgres = mkEnableOption "use local postgres instance";
```

Comment:
*IF* (and only if) it fits we have a bit of a pattern in NixOS to create a set of `database` options (yes, even when a `settings` option already contains database configuration. We usually add a `services.foo.database.createLocally` option and infer the details from the rest of the options. Take a look at a few other modules which do this if you're interested in being consistent with other modules in this regard: https://search.nixos.org/options?channel=21.05&from=0&size=50&sort=relevance&type=packages&query=createLocally

My underlying motivation: https://github.com/NixOS/nixpkgs/issues/62073